.kokoro/nightly/downstream-protobuf-source-compatibility.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eo pipefail
+
+# Comma-delimited list of repos to test with the local java-shared-dependencies
+if [ -z "${REPOS_UNDER_TEST}" ]; then
+  echo "REPOS_UNDER_TEST must be set to run downstream-protobuf-source-compatibility.sh"
+  exit 1
+fi
+
+# Version of Protobuf-Java runtime to compile with
+if [ -z "${PROTOBUF_RUNTIME_VERSION}" ]; then
+  echo "PROTOBUF_RUNTIME_VERSION must be set to run downstream-protobuf-source-compatibility.sh"
+  exit 1
+fi
+
+# Get the directory of the build script
+scriptDir=$(realpath "$(dirname "${BASH_SOURCE[0]}")")
+cd "${scriptDir}/../.." # cd to the root of this repo
+# common.sh lives in the presubmit folder
+source "$scriptDir/../presubmit/common.sh"
+setup_maven_mirror
```

Comment:
Is this still needed now that we are overriding the property, not building sdk-platform-java anymore?

Reply:
I don't think we need to call `setup_maven_mirror`, but I kept it here for consistency with the other jobs that need to pull deps from Maven Central

I think `setup_maven_mirror` can be moved to a common.sh file that doesn't solely exist in the presubmit folder, but I didn't want to make that part of this PR.

Reply:
What is the difference if we don't call `setup_maven_mirror`?

Reply:
I don't recall the original context of using the maven mirror, but I believe it was to help speed up the CI jobs by reducing the time it takes to download the JARs. I just did a search and it doesn't look like it's being used anymore (maybe this was removed recently?).

I'll remove it since it doesn't seem to be used anymore.